### PR TITLE
[HPI-1059] Add support for PR base images.

### PR DESCRIPTION
**Pull request by zenoando**
Add support for PR base images.

- Push images on PR open, push, re-open
- Add Makefile target and script runner

### CODE REVIEW

File: .github/workflows/pr-image.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  build:
+    name: Build and push image from PR
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```
Comment (schattingh) [inline .github/workflows/pr-image.yaml]: Can we update this to `v4`?  `v3` is deprecated.  May also be a good idea to do a drive-by update on `build.yaml` too please.

File: .github/workflows/pr-image.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Log in to ghcr
+        uses: docker/login-action@v2
```
Comment (schattingh) [inline .github/workflows/pr-image.yaml]: I think there is a newer version of this action too.

File: .github/workflows/pr-image.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build the image
+        run: make build
+
+      - name: Push PR image to the registry
```
Comment (schattingh) [inline .github/workflows/pr-image.yaml:29]: Is there anyway to manage the lifecycle of these images so that we don't keep them as long as we keep the released versions?

Comment (opsymcdevface) [reply .github/workflows/pr-image.yaml:29]: Will check that and follow up of what I find.

Comment (opsymcdevface) [reply .github/workflows/pr-image.yaml:29]: Used this [action](https://github.com/marketplace/actions/delete-package-versions) to delete all pre-release images.

Comment (opsymcdevface) [reply .github/workflows/pr-image.yaml:29]: Trying to have a deletion scheme only for the `prbase` images, so far am not confident with the delete-action setting I have in place.

**Review by schattingh — APPROVED** [review]
